dapp/src/components/AccountListener.js

```diff
@@ -34,10 +36,16 @@ const AccountListener = (props) => {
   const rates = useStoreState(StakeStore, (s) => s.rates)
   const prevRefetchStakingData = usePrevious(refetchStakingData)
   const prevRefetchUserData = usePrevious(refetchUserData)
-  const isDevelopment = process.env.NODE_ENV === 'development'
-  const isProduction = process.env.NODE_ENV === 'production'
   const AIR_DROPPED_STAKE_TYPE = 1
 
+  const balancesQuery = useBalancesQuery(account, contracts, {
+    onSuccess: (balances) => {
+      AccountStore.update((s) => {
+        s.balances = balances
```

Comment:
Right now we use the store (`AccountStore`) to access balances - but in the future we could actually just tap into react-query data. (essentially using `balancesQuery.data`).

Once all the async stuff move to react-query we can slowly migrate to just using it instead of the store.

Reply:
Sounds good to me. No need for a second global storage of the pulled data.